R/utils.R

```diff
@@ -810,12 +810,32 @@ split_scientific_notn <- function(x_str) {
 #' @noRd
 tidy_gsub <- function(x, pattern, replacement, fixed = FALSE) {
 
-  gsub(pattern, replacement, x, fixed = fixed)
+  if (!utf8_aware_sub) {
```

Comment:
Please add unit tests

```r
identical(tidy_sub(".", ".", "\u00B1", fixed = TRUE), "\u00B1")
identical(tidy_gsub(".", ".", "\u00B1", fixed = TRUE), "\u00B1")
```

Reply:
These tests were added (and tests for usage of `I()` were added as well).